controllers/argocd_metrics_controller.go

```diff
@@ -109,73 +109,135 @@ func (r *ArgoCDMetricsReconciler) Reconcile(ctx context.Context, request reconci
+		}
 
-	// Create alert rule
-	err = r.createPrometheusRuleIfAbsent(request.Namespace, argocd, reqLogger)
-	if err != nil {
-		return reconcile.Result{}, err
-	}
+		err = r.reconcileDashboards(reqLogger)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
 
-	err = r.reconcileDashboards(reqLogger)
-	if err != nil {
-		return reconcile.Result{}, err
-	}
+		err = r.reconcileOperatorMetricsServiceMonitor(reqLogger)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+	} else {
+		if exists {
+			namespace.Labels[clusterMonitoringLabel] = "false"
+			err = r.Client.Update(ctx, &namespace)
+			if err != nil {
+				reqLogger.Error(err, "Error updating namespace",
+					"Namespace", namespace.Name)
+				return reconcile.Result{}, err
+			}
 
-	err = r.reconcileOperatorMetricsServiceMonitor(reqLogger)
-	if err != nil {
-		return reconcile.Result{}, err
+			// Delete role to grant read permission to the openshift metrics stack
+			err = r.Client.Delete(context.TODO(), &rbacv1.Role{ObjectMeta: metav1.ObjectMeta{Namespace: request.Namespace, Name: fmt.Sprintf(readRoleNameFormat, request.Namespace)}})
+			if err != nil {
+				if !errors.IsNotFound(err) {
+					reqLogger.Error(err, "Error deleting role in ",
+						"Namespace", request.Namespace)
+					return reconcile.Result{}, err
+				}
+			}
+
+			// Delete role binding to grant read permission to the openshift metrics stack
+			err = r.Client.Delete(context.TODO(), &rbacv1.RoleBinding{ObjectMeta: metav1.ObjectMeta{Namespace: request.Namespace, Name: fmt.Sprintf(readRoleBindingNameFormat, request.Namespace)}})
+			if err != nil {
+				if !errors.IsNotFound(err) {
+					reqLogger.Error(err, "Error deleting rolebinding in ",
+						"Namespace", request.Namespace)
+					return reconcile.Result{}, err
+				}
+			}
+
+			// Delete ServiceMonitor for ArgoCD application metrics
+			serviceMonitorName := request.Name
+			err = r.deleteServiceMonitor(serviceMonitorName, request.Namespace, reqLogger)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+
+			// Delete ServiceMonitor for ArgoCD API server metrics
+			serviceMonitorName = fmt.Sprintf("%s-server", request.Name)
+			err = r.deleteServiceMonitor(serviceMonitorName, request.Namespace, reqLogger)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+
+			// Delete ServiceMonitor for ArgoCD repo server metrics
+			serviceMonitorName = fmt.Sprintf("%s-repo-server", request.Name)
+			err = r.deleteServiceMonitor(serviceMonitorName, request.Namespace, reqLogger)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+
+			// Delete alert rule
+			err = r.Client.Delete(context.TODO(), &monitoringv1.PrometheusRule{ObjectMeta: metav1.ObjectMeta{Namespace: request.Namespace, Name: alertRuleName}})
+			if err != nil {
+				if !errors.IsNotFound(err) {
+					reqLogger.Error(err, "Error deleting prometheus in ",
+						"Namespace", request.Namespace)
+					return reconcile.Result{}, err
+				}
+			}
```

Comment:
Are the prometheus rules deleted here the same as created by `.spec.monitoring.enabled` ?
https://argocd-operator.readthedocs.io/en/latest/usage/monitoring/

Reply:
No, this is different